lib/nerves_hub_link/client.ex

```diff
@@ -108,6 +108,22 @@ defmodule NervesHubLink.Client do
   @spec handle_fwup_message(fwup_message()) :: :ok
   def handle_fwup_message(data) do
     _ = apply_wrap(mod(), :handle_fwup_message, [data])
+
+    # TODO: nasty side effects here. Consider moving somewhere else
+    case data do
+      {:progress, percent} ->
+        _ = NervesHubLink.DeviceChannel.send_update_progress(percent)
+
+      {:error, _, message} ->
+        _ = NervesHubLink.DeviceChannel.send_update_status("fwup error #{message}")
+
+      {:ok, 0, _message} ->
+        _ = Nerves.Runtime.reboot()
```

Comment:
In `ssh_subsystem_fwup`, I had to run `Nerves.Runtime.reboot` in a separate process to avoid affecting shutdown of this GenServer. I just `spawn`'d `Nerves.Runtime.reboot` and nothing special.

Reply:
yeah i saw that in logs. Good suggestion, updating